hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/rest/api/QuestionsEntityResource.java

```diff
@@ -49,22 +51,26 @@ public class QuestionsEntityResource implements EntityResourceAction.Create<Ques
 {
     private final HxInsightClient hxInsightClient;
     private final QuestionsApiConfig questionConfig;
+    private final QuestionPermissionService questionPermissionService;
 
     @Override
     @WebApiDescription(title = "Ask question", successStatus = Status.STATUS_OK)
     public List<QuestionModel> create(List<QuestionModel> questions, Parameters parameters)
     {
         ensureThat(questions.size() == 1, () -> new WebScriptException(Status.STATUS_BAD_REQUEST, "You can only ask one question at a time."));
 
-        QuestionModel question = questions.get(0);
-
-        ensureThat(question.getRestrictionQuery().nodesIds().size() <= questionConfig.maxContextSizeForQuestion(),
-                () -> new WebScriptException(Status.STATUS_BAD_REQUEST, "You can only ask about up to %d nodes at a time.", questionConfig.maxContextSizeForQuestion()));
+        QuestionModel questionModel = questions.get(0);
+        Question question = questionModel.toQuestion();
 
         log.info("Received question: {}", question);
```

Comment:
You are not logging the original question, but the mapped object - maybe `questionModel` should be logged here?

Reply:
doesn't matter honestly